nexus/tests/integration_tests/instances.rs

```diff
@@ -238,13 +264,18 @@ async fn test_instances_create_reboot_halt(
     /*
      * Attempt to reboot the halted instance.  This should fail.
      */
-    let _error = client
-        .make_request_error(
-            Method::POST,
-            &format!("{}/reboot", instance_url),
-            StatusCode::BAD_REQUEST,
-        )
-        .await;
+    let _error: HttpErrorResponseBody = NexusRequest::expect_failure(
+        client,
+        StatusCode::BAD_REQUEST,
+        Method::POST,
+        &format!("{}/reboot", &instance_url),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .unwrap()
+    .parsed_body()
+    .unwrap();
```

Comment:
I don't see an obvious way around it besides adding yet another helper, but these chains are long and there are a lot of them

Reply:
I agree -- and a little error prone.  (It's easy to forget the `authn_as()` and get a 404 for the wrong reason.)  And you probably remember that the reason we _got_ here was we had like 5-10 helper functions that were all only slightly different from each other :-/

It's possible most of our tests should be using clients generated from the OpenAPI spec.  That could make a lot of these a lot clearer.

I've also been considering using something like [hurl](https://hurl.dev/) for our tests, where we use something like a DSL to describe the requests we want to make and the expected responses.  This sounds fantastic at first but it gets hairy when you add authn and stuff like the side-door operations used by the tests to manipulate Nexus state outside the API.